### PR TITLE
OpGroupWaitEvents doesn't require the Group capability

### LINKED ISSUE #595: OpGroupWaitEvents shouldn't require CapabilityGroups
Per the SPIR-V spec, OpGroupWaitEvents only requires the Kernel capability. But since it uses SPIRVGroupInstBase, it causes the resulting kernel to require Groups capability.

### PR DESCRIPTION
**Pull request by karolherbst**
Fixes #595

### CODE REVIEW

**Comment by AlexeySachkov** [comment]
Is it possible to add a test?

**Comment by karolherbst** [comment]
Sure, will do

File: test/event_no_group_cap.cl [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#include <opencl-c.h>
+
```
Comment (AlexeySachkov) [inline test/event_no_group_cap.cl:2]:
```suggestion
```

File: test/event_no_group_cap.cl [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+__kernel void test_fn( const __global char *src)
+{
+	wait_group_events(0, NULL);
+}
+// RUN: %clang_cc1 -triple spir64 -x cl -cl-std=CL2.0 -O0 -emit-llvm-bc %s -o %t.bc
```
Comment (AlexeySachkov) [inline test/event_no_group_cap.cl:7]:
```suggestion
// RUN: %clang_cc1 -triple spir64 -x cl -cl-std=CL2.0 -fdeclare-opencl-builtins -O0 -emit-llvm-bc %s -o %t.bc
```

**Review by AlexeySachkov — COMMENTED** [review]
LGTM

A few minor comments regarding the test. Also, @karolherbst, could you please rebase your PR to get CI checks passed?

**Review by svenvh — APPROVED** [review]

**Pull request merged**